src/rf/js/src/home/components/library.js

```diff
@@ -356,7 +356,8 @@ var LayerItem = React.createBackboneClass({
     },
 
     deleteLayer: function() {
-        console.log('delete layer');
+        this.getModel().destroy();
+        settings.getPendingLayers().remove(this.getModel());
```

Comment:
It would be nice if a single call to destroy removed the layer from both collections. I'm not sure how to achieve this though.